apps/settings/js/modules/bluetooth/bluetooth_context.js

```diff
@@ -824,6 +844,221 @@ define(function(require) {
         Debug('unpair(): Reject with this reason: ' + reason);
         return Promise.reject(reason);
       });
+    },
+
+    /**
+     * Init the connecting device which is browsed in paired devices list. 
+     * Get connection info from ConnectionManager.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     */
+    _initConnectingDevices: function btc__initConnectingDevices() {
+      // Init the paired device connection status for connecting device.
+      if (!ConnectionManager.connectingAddress) {
+        return;
+      }
+
+      var index = this.getPairedDevices().array.findIndex(
```

Comment:
Let's just wrap the code in a single function that returns the device item of a specific address as you use it a lot in other functions. `_findDeviceByAddress` is a good name to me.